pkg/yaml/yaml_test.go

```diff
@@ -0,0 +1,496 @@
+	if nonExistingEmptyYamlErr == nil {
+		t.Errorf("Should throw an error because %s doesn't exist", nonExistingEmptyYamlFileName)
+	}
+}
+
+func TestReadYamlAllFileFs(t *testing.T) {
+	// Setup variables
+	existingEmptyYamlFileNamePattern := "test_read_yaml_all_file_existing_empty_*.yaml"
+	nonExistingEmptyYamlFileName := "nonExistingEmpty.yaml"
+
+	// Create directory for test files
+	createTestDir()
+	defer cleanupTestDir()
+
+	// Setup temporary file
+	file, err := CreateTempFile(existingEmptyYamlFileNamePattern)
+	defer os.Remove(file.Name())
+	if err != nil {
+		t.Errorf("Can't create file: %s", file.Name())
+	}
+
+	//Read existing empty yaml file
+	existingEmptyYamlAllFileResult, existingEmptyYamlAllFileErr := ReadYamlAllFile(file.Name())
+	if existingEmptyYamlAllFileErr != nil {
+		t.Errorf("Can't read empty yaml file: %s", file.Name())
+	}
+
+	if existingEmptyYamlAllFileResult != nil {
+		t.Errorf("Empty YAML stream read incorrectly. Value should be nil")
+	}
+
+	//Read non-existing empty yaml file
+	nonExistingEmptyYamlAllFileResult, nonExistingEmptyYamlAllFileErr := ReadYamlAllFile(nonExistingEmptyYamlFileName)
+	if nonExistingEmptyYamlAllFileErr == nil {
+		t.Errorf("Should throw an error because %s doesn't exist", nonExistingEmptyYamlFileName)
+	}
+
+	if nonExistingEmptyYamlAllFileResult != nil {
+		t.Errorf("Empty YAML stream read incorrectly. Value should be nil because file doesn't exist.")
+	}
+}
+
+func TestReadYamlStream(t *testing.T) {
+	// Setup variables
+	simpleYamlDataString := `
+    value: anyValue
+    `
+	simpleYamlDataReader := bytes.NewReader([]byte(simpleYamlDataString))
+	var existingSimpleYamlConf SimpleYamlConfig
+
+	existingReadYamlStreamErr := ReadYamlStream(simpleYamlDataReader, &existingSimpleYamlConf)
+	if existingReadYamlStreamErr != nil {
+		t.Errorf("Can't read simple yaml stream: %s", existingReadYamlStreamErr)
+	}
+
+	//Check if it can handle errors
+	errorReadYamlStreamErr := ReadYamlStream(iotest.ErrReader(errors.New("timeout")), &EmptyYamlConfig{})
+	if errorReadYamlStreamErr == nil {
+		t.Errorf("It should throw an error because of a timeout")
+	}
+}
+
+func TestReadYamlString(t *testing.T) {
+	// Setup variables
+	simpleYamlDataString := `
+    value: anyValue
+    `
+	var existingSimpleYamlConf SimpleYamlConfig
+
+	existingReadYamlStringErr := ReadYamlString(simpleYamlDataString, &existingSimpleYamlConf)
+	if existingReadYamlStringErr != nil {
+		t.Errorf("Can't read simple yaml stream: %s", existingReadYamlStringErr)
+	}
+}
+
+func TestReadYamlAllString(t *testing.T) {
+	// Setup variables
+	simpleYamlDataString := `
+    value: anyValue
+    `
+	simpleYamlAllStringResult, simpleYamlAllStringErr := ReadYamlAllString(simpleYamlDataString)
+	if simpleYamlAllStringErr != nil {
+		t.Errorf("Can't read simple yaml stream: %s", simpleYamlAllStringErr)
+	}
+	simpleYamlAllStringResultMap := simpleYamlAllStringResult[0].(map[string]interface{})
+	if simpleYamlAllStringResultMap["value"] != "anyValue" {
```

Comment:
```go
assert.Equal(t, []map[string]any{
  map[string]any{
    "value:" "anyValue",
  },
}, simpleYamlAllStringResult)
```

Reply:
Done in e4c1c968286ad5a125bcbd551c6f9775e0dd4707